### PR TITLE
Price aggregator integration

### PR DESCRIPTION
**Pull request by dorin-iancu**
Integrate the Ethereum tx fees price aggregator into the EthereumFeePrepay and (indirectly) into EsdtSafe + tests

### CODE REVIEW

File: ethereum-fee-prepay/src/lib.rs [hunk 1]
```diff
@@ -46,63 +48,40 @@ pub trait EthereumFeePrepay {
...
 
     #[endpoint(payFee)]
-    fn pay_fee(
-        &self,
-        address: Address,
-        relayer: Address,
-        transaction_type: TransactionType,
-        priority: Priority,
-    ) -> SCResult<()> {
+    fn pay_fee(&self, tx_senders: Vec<Address>, relayer: Address) -> SCResult<()> {
```
Comment (sasurobert) [inline ethereum-fee-prepay/src/lib.rs:51]: can we extend this to accept a series of whitelisted tokens ? maybe we want ETH here.

File: ethereum-fee-prepay/src/lib.rs [hunk 2]
```diff
@@ -46,63 +48,40 @@ pub trait EthereumFeePrepay {
...
-        };
+            .latest_price_feed(BoxedBytes::from(GWEI_STRING), BoxedBytes::from(EGLD_STRING))
+            .execute_on_dest_context()
+            .into();
 
-        Ok(aggregator_result.egld_to_eth * gas_limit * gas_price
-            / aggregator_result.egld_to_eth_scaling)
+        self.last_query_price().set(&aggregator_result.price);
+
+        aggregator_result.price * ETH_ERC20_TX_GAS_LIMIT.into()
```
Comment (mateigraura) [inline ethereum-fee-prepay/src/lib.rs:84]: i don't think this is correct. multiplying gas limit with a 18 decimal based denomination (EGLD) will result in precision errors. I'm not sure how this is intended to work, but if you wish to estimate the tx cost in Ethereum, using the GWEI to EGLD denomination might not be the way to go.

**Review by mateigraura — CHANGES_REQUESTED** [review]

File: ethereum-fee-prepay/src/lib.rs (hunk 2 above)
Comment (mateigraura) [reply ethereum-fee-prepay/src/lib.rs:84]: tested and the relative value in USD is equivalent. closing.

File: ethereum-fee-prepay/src/lib.rs (hunk 1 above)
Comment (dorin-iancu) [reply ethereum-fee-prepay/src/lib.rs:51]: Done on another branch.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**